include/nbl/builtin/hlsl/device_capabilities_traits.hlsl

```diff
@@ -7,15 +7,1323 @@
+
+NBL_GENERATE_GET_OR_DEFAULT(lineSubPixelPrecisionBits, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat16Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat16AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat16AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferFloat64AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat16Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat16AtomicAdd, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat16AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedFloat64AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderImageFloat32AtomicMinMax, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(sparseImageFloat32AtomicMinMax, bool, false);
+
+
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayout, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayoutScalarBlockLayout, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayout8BitAccess, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(workgroupMemoryExplicitLayout16BitAccess, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(colorWriteEnable, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(cooperativeMatrixSupportedStages, core::bitflag<asset::IShader::E_SHADER_STAGE>, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderBufferInt64Atomics, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSharedInt64Atomics, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderFloat16, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderInputAttachmentArrayDynamicIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderUniformBufferArrayNonUniformIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderInputAttachmentArrayNonUniformIndexing, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(descriptorBindingUniformBufferUpdateAfterBind, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(samplerFilterMinmax, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(vulkanMemoryModelAvailabilityVisibilityChains, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderOutputViewportIndex, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderOutputLayer, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderDemoteToHelperInvocation, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderTerminateInvocation, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(shaderZeroInitializeWorkgroupMemory, bool, false);
+
+//Nabla
+NBL_GENERATE_GET_OR_DEFAULT(computeUnits, uint32_t, false);
+NBL_GENERATE_GET_OR_DEFAULT(dispatchBase, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(allowCommandBufferQueryCopies, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxOptimallyResidentWorkgroupInvocations, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxResidentInvocations, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(spirvVersion, asset::CGLSLCompiler::E_SPIRV_VERSION, 0);
```

Comment:
we might need to make that enum global and then alias or something?

Reply:
I'm not sure what you mean here. Can't we just replace the enum with an integral type like the rest?